docs: Add .Net caching to docs (#3839)
Flagsmith/flagsmith: docs/docs/clients/server-side.md

    # Override the default Flagsmith API URL if you are self-hosting.
    # Optional.
    # Defaults to https://edge.api.flagsmith.com/api/v1/
    apiUrl: "https://flagsmith.myproject.com",

>     # Enables caching support.

Reviewer: This should be lower down in the config options - follow the other languages that have caching

final FlagsAndTraits flags = cache.getIfPresent(projectLevelCacheKey);
```

</TabItem>
<TabItem value="dotnet" label=".NET">

You can initialise the SDK like this:

```csharp
            var flagsmithClient = new FlagsmithClient(Fixtures.ApiKey,
                httpClient: mockHttpClient.Object,
                enableClientSideEvaluation: true,
                cacheConfig: new CacheConfig(true));
```

> If cacheConfig is passed during initialization with its `Enabled` property set to `true`, the SDK will make use cached

Reviewer: ```suggestion
If cacheConfig is passed during initialization with its `Enabled` property set to `true`, the SDK will make use of cached
```

values initially, and then make required API calls as usual and update the cache under the hood.

> Currently the TTL for the cache is hard set to 5 minutes.

Reviewer: ```suggestion
The TTL for the cache is 5 minutes, and is not configurable.
```
